apps/ledger-live-desktop/src/renderer/families/elrond/StakeBanner.tsx

```diff
@@ -0,0 +1,92 @@
+import { useTranslation } from "react-i18next";
+import useFeature from "@ledgerhq/live-common/featureFlags/useFeature";
+
+import { getAccountBannerState as getElrondBannerState } from "@ledgerhq/live-common/families/elrond/banner";
+import { AccountBanner } from "~/renderer/screens/account/AccountBanner";
+import React from "react";
+import { StakeAccountBannerParams } from "~/renderer/screens/account/types";
+import { useElrondPreloadData } from "@ledgerhq/live-common/families/elrond/react";
+import { ElrondAccount } from "@ledgerhq/live-common/families/elrond/types";
+import { track } from "~/renderer/analytics/segment";
+import { stakeDefaultTrack } from "~/renderer/screens/stake/constants";
+import { useDispatch } from "react-redux";
+import { openModal } from "~/renderer/actions/modals";
+import { modals } from "~/renderer/families/elrond/modals";
+import { AccountBannerState } from "@ledgerhq/live-common/lib/families/elrond/banner";
+
+export const StakeBanner: React.FC<{ account: ElrondAccount }> = ({ account }) => {
+  const { t } = useTranslation();
+  const stakeAccountBanner = useFeature("stakeAccountBanner");
+  const stakeAccountBannerParams: StakeAccountBannerParams | null =
+    stakeAccountBanner?.params ?? null;
+
+  const elrondPreloadData = useElrondPreloadData();
+  const dispatch = useDispatch();
+  const bannerState: AccountBannerState = getElrondBannerState(account, elrondPreloadData);
+  const showDelegationBanner = bannerState.bannerType === "delegate";
```

Comment:
Can we unify the criteria across all the banners?
(I don't mind use 'bannerType' and 'bannerState', but I think what it is more important is have a "common" approach)
```
const { redelegate, display, ...} = state;

Reply:
Yeah I agree, a common approach would be nice.

I did it this way as you get better type safety with a discriminated union type, i.e. you don't need optional params on `bannerState` for the redelegate fields. I'd be happy to refactor the others to keep it consistent if you're happy with the approach

Reply:
(But please use the destructuring syntax)

Reply:
If it's okay with you, I'm going to refactor in a separate PR to minimise the changes in this PR.